kodi-cli: Init at 1.1.1

pstn opened the pull request:
###### Motivation for this change
Added the handy utility `kodi-cli` for controlling remote kodi instances with a cli.
I was on the fence myself about PRing this, so I asked in #nixos. A few people wanted to have it, so here it is.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

pstn committed:
kodi-cli: Init at 1.1.1

worldofpeace reviewed (changes requested):
A lot of the indentation and whitespace isn't up to par with what's in nixpkgs. 
See: https://nixos.org/nixpkgs/manual/#sec-syntax

worldofpeace commented on pkgs/applications/video/kodi-cli/default.nix:
```diff
@@ -0,0 +1,35 @@
+in
+
+stdenv.mkDerivation rec {
+  name = pkgName + "-" + version;
```
Something more popular in nixpkgs would be
```suggestion
  name = "kodi-cli-${version}";
```

And move the `version = "1.1.1";` into the set doing away with the `let in`.

worldofpeace commented on pkgs/applications/video/kodi-cli/default.nix:
```diff
@@ -0,0 +1,35 @@
+    description = "Kodi/XBMC bash script to send Kodi commands using JSON RPC. It also allows sending YouTube videos to Kodi";
+    license = licenses.gpl2;
+    platforms = platforms.all;
+    maintainers = [ maintainers.pstn ];
```
You'll need to add yourself to the https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix

pstn replied:
I'm already in there somewhere. Wouldn't @GrahamcOfBorg complain otherwise btw?

symphorien commented on pkgs/applications/video/kodi-cli/default.nix:
```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, makeWrapper, pkgs, curl, bash, jq, youtube-dl, gnome3, ...}:
```
Since https://github.com/NixOS/nixpkgs/pull/49398 it is enough to write
```
pname = "kodi-cli";
version = "1.1.1";
```

worldofpeace replied:
@symphorien Oh, I wasn't aware that hit master yet.

pstn replied:
It's merged but only in the `name` version, not in the `pname` version.
When I use `name`, my package builds fine, but with `pname` it breaks.
I suggest to just leave this unmerged, until the other change is merged and not add a package that will break again in a few days.

pstn commented:
Whoopsie. Made a little git mess. Please disregard until I clean it up.

worldofpeace commented on pkgs/applications/video/kodi-cli/default.nix:
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, makeWrapper, pkgs, curl, bash, jq, youtube-dl, gnome3 }:
+
+stdenv.mkDerivation rec {
+  version = "1.1.1";
```
The changes @symphorien was speaking of for the derivation names isn't in master yet so this should be

`name = "kodi-cli-${version};`
`version = "1.1.1";`

We'll probably just do a big refactor of this anyway so there isn't a need to constrain this.

c0bw3b commented on pkgs/applications/video/kodi-cli/default.nix:
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, makeWrapper, pkgs, curl, bash, jq, youtube-dl, gnome3 }:
```
```suggestion
{ stdenv, fetchFromGitHub, makeWrapper, curl, bash, jq, youtube-dl, gnome3 }:
```

c0bw3b commented on pkgs/applications/video/kodi-cli/default.nix:
```diff
@@ -0,0 +1,30 @@
+  };
+
+
+  buildInputs = [ makeWrapper ];
```
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

pstn replied:
Oh, yeah. I always confuse those.

c0bw3b commented on pkgs/applications/video/kodi-cli/default.nix:
```diff
@@ -0,0 +1,30 @@
+    cp -a kodi-cli $out/bin
+    wrapProgram $out/bin/kodi-cli --prefix PATH : ${stdenv.lib.makeBinPath [ curl bash ]}
+    cp -a playlist_to_kodi $out/bin
+    wrapProgram $out/bin/playlist_to_kodi --prefix PATH : ${stdenv.lib.makeBinPath [ curl bash gnome3.zenity jq youtube-dl]}
```
```suggestion
    wrapProgram $out/bin/playlist_to_kodi --prefix PATH : ${stdenv.lib.makeBinPath [ curl bash gnome3.zenity jq youtube-dl ]}
```
(nitpick)

pstn replied:
Rightly so :+1:

pstn commented on pkgs/applications/video/kodi-cli/default.nix:
```diff
@@ -0,0 +1,30 @@
+    license = licenses.gpl2;
+    platforms = platforms.all;
+    maintainers = [ maintainers.pstn ];
+};
```
Seems like my indentation plugin is buggy. Need to fix that I guess.

worldofpeace committed:
kodi-cli: nitpicks (please fixup)

nitpicks applied are:

  - The pname thing
    staging-next has been merged.

  - Moved to tools/misc
    applications/video is more appropriate for video applications.
    This is a script used to interact with one.

  - Changed platforms to unix
    This script can only be used where kodi is present.

worldofpeace commented:
@pstn I've pushed basically reviewer nitpicks to your branch. I hope you don't mind.

Also, my apologies for your change being shelved on this `pname` discussion.
IMHO that shouldn't have been important as that change wasn't even on master yet.

pstn commented:
@worldofpeace All good! I actually made the same point about waiting with a merge for this package myself. I also like the nitpicks, just had a busy week and no time to merge.
